src/server/qgsserverprojectparser.cpp

```diff
@@ -644,69 +644,6 @@ QString QgsServerProjectParser::layerName( const QDomElement& layerElem ) const
   return nameElem.text().replace( QLatin1String( "," ), QLatin1String( "%60" ) ); //commas are not allowed in layer names
 }
 
-QString QgsServerProjectParser::serviceUrl() const
-{
-  QString url;
-
-  if ( !mXMLDoc )
-  {
-    return url;
-  }
-
-  QDomElement propertiesElement = propertiesElem();
-  if ( !propertiesElement.isNull() )
-  {
-    QDomElement wmsUrlElem = propertiesElement.firstChildElement( QStringLiteral( "WMSUrl" ) );
-    if ( !wmsUrlElem.isNull() )
-    {
-      url = wmsUrlElem.text();
-    }
-  }
-  return url;
-}
-
-QString QgsServerProjectParser::wfsServiceUrl() const
-{
-  QString url;
-
-  if ( !mXMLDoc )
-  {
-    return url;
-  }
-
-  QDomElement propertiesElement = propertiesElem();
-  if ( !propertiesElement.isNull() )
-  {
-    QDomElement wfsUrlElem = propertiesElement.firstChildElement( QStringLiteral( "WFSUrl" ) );
-    if ( !wfsUrlElem.isNull() )
-    {
-      url = wfsUrlElem.text();
-    }
-  }
-  return url;
-}
-
-QString QgsServerProjectParser::wcsServiceUrl() const
-{
-  QString url;
-
-  if ( !mXMLDoc )
-  {
-    return url;
-  }
-
-  QDomElement propertiesElement = propertiesElem();
-  if ( !propertiesElement.isNull() )
-  {
-    QDomElement wcsUrlElem = propertiesElement.firstChildElement( QStringLiteral( "WCSUrl" ) );
-    if ( !wcsUrlElem.isNull() )
-    {
-      url = wcsUrlElem.text();
-    }
-  }
-  return url;
-}
-
 void QgsServerProjectParser::combineExtentAndCrsOfGroupChildren( QDomElement& groupElem, QDomDocument& doc, bool considerMapExtent ) const
```

Comment:
Actually the build of service URL is done on a per-service basis from the requests url (no more getenv access). There is no exposed service url atm in QgsService interface but that can be easily added. 
See  https://github.com/qgis/QGIS/blob/master/src/server/services/wms/qgswmsutils.h#L56. This function is basically implemented for each services.